static/js/drafts.js

```diff
@@ -148,6 +148,12 @@ exports.restore_draft = function (draft_id) {
     }
     compose_actions.start(draft_copy.type, draft_copy);
     compose_ui.autosize_textarea();
+    if ($("#preview_message_area").is(":visible") && $("#undo_markdown_preview").is(":visible")) {
+        $("#markdown_preview").show();
+        $("#undo_markdown_preview").hide();
+        $("#preview_message_area").hide();
+        $("#new_message_content").show();
+    }
```

Comment:
This fix is likely correct, but it's unlikely to stay correct as time goes on and we maintain the codebase.  We should refactor the compose code that we can have this case call the same function the compose code uses to switch out of preview mode.